bundledata.go

```diff
@@ -455,6 +459,29 @@ func (verifier *bundleDataVerifier) verifyRelations() {
 	}
 }
 
+func (verifier *bundleDataVerifier) verifyEndpointBindings() {
+	for name, svc := range verifier.bd.Services {
+		charm, ok := verifier.charms[name]
+		// Only thest the ok path here because the !ok path is tested in verifyServices
+		if !ok {
+			continue
+		}
+		for endpoint, space := range svc.EndpointBindings {
+			_, matchedProvides := charm.Meta().Provides[endpoint]
+			_, matchedRequires := charm.Meta().Requires[endpoint]
+			_, matchedPeers := charm.Meta().Peers[endpoint]
+
+			if !(matchedProvides || matchedRequires || matchedPeers) {
+				verifier.addErrorf(
+					"service %s wants to bind to endpoint %s to space %s, "+
+						"which isn't provided or required by charm",
```

Comment:
Strictly speaking, "provided or required" is misleading, as it also checks "peers". So how about "isn't defined by the charm" ?